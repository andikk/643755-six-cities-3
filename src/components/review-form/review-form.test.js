import React from "react";
import renderer from "react-test-renderer";
import ReviewForm from "./review-form.jsx";

it(`Should ReviewForm render correctly`, () => {

  const tree = renderer
    .create(<ReviewForm
      offerId = {1}
      handleFormSubmit = {() => {}}
      rating = {1}
      comment = {`PropTypes.string`}
      invalid = {true}
      submitting = {true}
      onRatingChange = {() => {}}
      onCommentChange = {() => {}}
      onSubmit = {() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
