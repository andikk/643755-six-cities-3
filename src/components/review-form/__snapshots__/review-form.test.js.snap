// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewForm render correctly 1`] = `
<form
  action=""
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="5-star"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={5}
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-star"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="4-star"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={4}
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-star"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="3-star"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={3}
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-star"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="2-star"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={2}
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-star"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="1-star"
      name="rating"
      onChange={[Function]}
      type="radio"
      value={1}
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-star"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value=""
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled={true}
      style={
        Object {
          "color": "#9c9c9c",
        }
      }
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
